external/anomaly/tests/test_ote_anomaly_segmentation_training.py

```diff
@@ -0,0 +1,420 @@
+
+    def __call__(self, data_collector: DataCollector, results_prev_stages: OrderedDict):
+        self._run_ote_training(data_collector)
+        results = {
+            "model_template": self.model_template,
+            "task": self.task,
+            "dataset": self.dataset,
+            "environment": self.environment,
+            "output_model": self.output_model,
+        }
+        return results
+
+
+class TestOTEReallifeAnomalySegmentation(OTETrainingTestInterface):
+    """
+    The main class of running test in this file.
+    """
+    PERFORMANCE_RESULTS = None # it is required for e2e system
+    helper = OTETestHelper(AnomalySegmentationTrainingTestParameters())
+
+    @classmethod
+    def get_list_of_tests(cls, usecase: Optional[str] = None):
+        """
+        This method should be a classmethod. It is called before fixture initialization, during
+        tests discovering.
+        """
+        return cls.helper.get_list_of_tests(usecase)
+
+    @pytest.fixture
+    def params_factories_for_test_actions_fx(self, current_test_parameters_fx,
+                                             dataset_definitions_fx,ote_current_reference_dir_fx,
+                                             template_paths_fx) -> Dict[str,Callable[[], Dict]]:
+        logger.debug('params_factories_for_test_actions_fx: begin')
+
+        test_parameters = deepcopy(current_test_parameters_fx)
+        dataset_definitions = deepcopy(dataset_definitions_fx)
+        template_paths = deepcopy(template_paths_fx)
+        def _training_params_factory() -> Dict:
+            if dataset_definitions is None:
+                pytest.skip('The parameter "--dataset-definitions" is not set')
+            model_name = test_parameters['model_name']
+            dataset_name = test_parameters['dataset_name']
+            patience = test_parameters['patience']
+            batch_size = test_parameters['batch_size']
+            dataset_params = _get_dataset_params_from_dataset_definitions(dataset_definitions, dataset_name)
+
+            if model_name not in template_paths:
+                raise ValueError(f'Model {model_name} is absent in template_paths, '
+                                 f'template_paths.keys={list(template_paths.keys())}')
+            template_path = make_path_be_abs(template_paths[model_name], template_paths[ROOT_PATH_KEY])
+            logger.debug('training params factory: Before creating dataset and labels_schema')
+            dataset, labels_schema = _create_anomaly_segmentation_dataset_and_labels_schema(dataset_params, dataset_name)
+            logger.debug('training params factory: After creating dataset and labels_schema')
+            return {
+                'dataset': dataset,
+                'labels_schema': labels_schema,
+                'template_path': template_path,
+                'patience': patience,
+                'batch_size': batch_size,
+            }
+
+        def _nncf_graph_params_factory() -> Dict:
```

Comment:
Looks like almost the same variables are returned as in `params_factories_for_test_actions_fx`. Perhaps we could use something similar `Dict[str,Callable[[], Dict]]` rather than just `Dict`?

Reply:
Corrected